### PR TITLE
Generate XML status report of the last hardlink backup run

### PR DESCRIPTION
**Pull request by individual-it**
added a function that generates a status report of the last backup run.
This can be used to determine the backup status of the computer (for inventory system, display a message to the user, notify the administrator)

@phil-davis please review. 
Should we use a new parameter instead of $emailJobName. "emailJobName" might be confusing as its not used not only for emails.

### CODE REVIEW

**Comment by phil-davis** [comment]
Yes, emailJobName should always have just been JobName. It is a general string that can be used to identify different tasks/jobs from each other. But for backward-compatibility you will need to keep emailJobName as an undocumented parameter that is copied into JobName if JobName is not specified. Then existing config files and command lines will keep working.

**Comment by phil-davis** [comment]
There could also be a parameter StatusFile that can specify the status file. That way when there are multiple backup jobs on 1 system the status output can be made unique, even if the jobs are writing to the same LogFile directory.
StatusFile could work like:
1) If just a filename[.type] is given then write it in the log file directory with the given name
2) If a folder is given and no filename[.type] then write status.xml in the specified folder
3) If a folder and file name[.type] is given then write it as specified
4) If StatusFile is not specified then do as now - LogFile folder with name status.xml

File: ntfs-hardlink-backup/ntfs-hardlink-backup.ps1 [hunk 1]
```diff
@@ -85,10 +85,11 @@
 	Switch off the use of Shadow Copies. Can be useful if you have no permissions to create Shadow Copies.
 .PARAMETER SMTPPort
 	Port of the SMTP Server. Default=587
-.PARAMETER emailJobName
-	This is added in to the auto-generated email subject "Backup of: hostname emailJobName by: username"
+.PARAMETER jobName
+	This is added in to the auto-generated email subject "Backup of: hostname jobName by: username" and used in the status file.
+	Using the 'emailJobName' parameter is depreciated
```
Comment (phil-davis) [inline ntfs-hardlink-backup/ntfs-hardlink-backup.ps1]: s/depreciated/deprecated/
"depreciated" is a finance word :)

File: ntfs-hardlink-backup/ntfs-hardlink-backup.ps1 [hunk 2]
```diff
@@ -790,6 +793,21 @@ if ([string]::IsNullOrEmpty($emailSubject)) {
 
 if ([string]::IsNullOrEmpty($emailJobName)) {
 	$emailJobName = Get-IniParameter "emailJobName" "${FQDN}"
+	
+	$output = "`WARNING: using the 'emailJobName' parameter is depreciated! Please use 'jobName'`n"
```
Comment (phil-davis) [inline ntfs-hardlink-backup/ntfs-hardlink-backup.ps1]: Another depreciated

**Comment by phil-davis** [comment]
Looks good and has flexibility.
:+1:

**Comment by phil-davis** [comment]
And bump the version - I guess this is 2.2-ALPHA1 ?

**Comment by phil-davis** [comment]
I can make the same enhancements to robocopy-backup. It needs to have consistent JobName parameter anyway, and I guess it can have StatusFile so that it could be monitored on the servers?

**Comment by individual-it** [comment]
depends if we would like to have more features in 2.2
if yes I would say call it ALPHA
if not then BETA for some testing time

we are planing to use the statusFile for OCS inventory & to remind the user if he haven't done a backup for a long time

**Comment by phil-davis** [comment]
Unless you have other features you are about to add, then make it BETA.

**Pull request merged**